presto-main/src/main/java/io/prestosql/spiller/AesSpillCipher.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spiller;
+
+import io.prestosql.spi.PrestoException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+
+final class AesSpillCipher
+        implements SpillCipher
+{
+    //  256-bit AES CBC mode
+    private static final String CIPHER_NAME = "AES/CBC/PKCS5Padding";
+    private static final int KEY_BITS = 256;
+
+    private SecretKey key;
+    //  Instance used only for determining encrypted output lengths
+    private Cipher encryptSizer;
+    private final int ivBytes;
+
+    AesSpillCipher()
+    {
+        this.key = generateNewSecretKey();
+        this.encryptSizer = initializeForEncrypt(createUninitializedCipher(), key);
+        this.ivBytes = encryptSizer.getIV().length;
+    }
+
+    @Override
+    public int encryptedMaxLength(int inputLength)
+    {
+        checkArgument(inputLength >= 0, "inputLength must be >= 0, inputLength=%s", inputLength);
+        return ivBytes + throwCipherClosedIfNull(encryptSizer).getOutputSize(inputLength);
```

Comment:
I think it would be more readable if you added an instance `checkNotClosed()` method to this class and then call that after the `checkArgument`s of each public method... specifically, I find the passthrough check in the middle of a more complex statement a bit hard to follow.

Reply:
The problem is that a `checkNotClosed()` method would still make it possible for a concurrent `close()` call to null the field between the check and subsequent usage. It seemed preferable to avoid adding synchronizing calls just for that case and instead do a null check + local variable usage pattern. I can separate the check from the call though if that's an improvement on readability, ie: 
```java
  Cipher sizer = throwCipherClosedIfNull(encryptSizer);
  return ivBytes + sizer.getOutputSize(inputLength);
}

Reply:
Is this class thread safe?  I didn't think it was.  If so, can you mark it with `@ThreadSafe`?  BTW, I missed this reply, due to the "outdated" github thing, when writing my review comments below.